File: Unit2-06/Jet.ts
/**
 * This is the Jet class.
 *
 * By: Flynn Rundquist
 * Version: 1.0
 * Since: 2024-04-09
 */

import Airplane from './Airplane'
class Jet extends Airplane {

    // accelerate
    public accelerate(speed: number) {
        this.speed = speed * 2
        return this.speed
    }
}
export default Jet

File: Unit2-06/Jet.java
/**
 * This class creates a Jet object.
 *
 * @author Flynn Rundquist
 * @version 1.0
 * @since 2024-04-24
 */
public class Jet extends Airplane {

    private static final int MULTIPLIER = 2;

    public Jet() {
        super();
    }

    public void setSpeed(int speed) {
        super.setSpeed(speed * MULTIPLIER);
    }

    public void accelerate() {
        setSpeed(getSpeed() * 2);
    }
}
